### ISSUE TITLE
News RSS button shows up on every new tab, News follow/unfollow modal remains open when rss button is removed after page load

### ISSUE THREAD

**Issue by srirambv** [brave/brave-browser#27727]
## Description 

News RSS button shows up on every `target="_blank"` tab, opening the RSS feed modal remains when rss feed button is removed

Note from @petemill : The main issue here is that the button thinks there are feeds for a Tab before the Tab gets an actual Url. The Url at that time is empty and it chooses the first directly-followed Publisher. Therefore the steps are simpler:
- Follow any publisher directly (e.g. ESPN as per the steps below)
- Open a New Tab Page
Observe: The News button flashes on screen briefly
Expected: The News button should never show for a New Tab.

## Steps to Reproduce

   1. Enable News 
   2. Visit `https://espn.com/football` 
   3. Click on the RSS feed button and follow it
   4. Click on a link that opens in a new tab
   5. RSS feed link shows up before page starts to load
   6. Click on the RSS feed button, shows ESPN Unfollow  button
   7. RSS button disappears when page loads but the modal remains open

## Actual result:

![image](https://user-images.githubusercontent.com/17010094/211623192-10d6c2e3-48e4-40f2-8f2a-6b87212e3d2d.png)

## Expected result:
Not show RSS feed when loading a page in a new tab 

## Reproduces how often: 

Easy

## Brave version (brave://version info)

<html>
<body>

Brave | 1.49.5 Chromium: 109.0.5414.80 (Official Build) nightly (64-bit)
-- | --
Revision | `0f69b168d36a06cace4365e9f029fa987afa5633-refs/branch-heads/5414@{#1178}`
OS | Windows 11 Version 22H2 (Build 22621.963)

</body>
</html>
## Version/Channel Information:

- Can you reproduce this issue with the current release? Couldn't repro
- Can you reproduce this issue with the beta channel? Couldn't repro
- Can you reproduce this issue with the nightly channel? Yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? NA
- Does the issue resolve itself when disabling Brave Rewards? NA
- Is the issue reproducible on the latest version of Chrome? NA

## Miscellaneous Information:

cc: @petemill @mattmcalister @brave/qa-team @rebron

**Comment by stephendonner** [comment]
Also cc @fallaciousreasoning

**Comment by MadhaviSeelam** [comment]
Verification `PASSED` using
```
Brave | 1.49.117 Chromium: 111.0.5563.50 (Official Build) (64-bit)
-- | --
Revision | 39f3dfc5023fe60997adc4f6cf5439cfa3e6968a-refs/branch-heads/5563@{#903}
OS | Windows 11 Version 22H2 (Build 22621.1265)
```
Verified using STR from https://github.com/brave/brave-browser/issues/27727#issue-1527776863

Confirmed RSS button does not show up on every new tab

https://user-images.githubusercontent.com/98358127/222822956-cea9f607-c52c-4140-b640-f03265887d3c.mp4

**Comment by srirambv** [comment]
Seems fixed can't repro it on `1.49.118`
